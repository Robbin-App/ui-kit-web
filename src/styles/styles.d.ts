import 'styled-components';
import { IBaseRobbinTheme } from './theme';

declare module 'styled-components' {
  export type DefaultTheme = IBaseRobbinTheme;
}
