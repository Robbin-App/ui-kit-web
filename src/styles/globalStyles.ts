import { createGlobalStyle, css } from 'styled-components';
import { normalize } from 'styled-normalize';

export const globalStyles = css`
  ${normalize}
  * {
    box-sizing: border-box;
    font-feature-settings: "lnum";
    &:focus {
      outline: 1px solid ${(props) => props.theme.colors.secondary100};
    }
  }
  body {
    font-family: 'Nunito', sans-serif;;
    color: ${(props) => props.theme.colors.text.globalText}
    background: ${(props) => props.theme.colors.greySurface};
    font-size: 16px;
  }
  html,
  body {
    overflow-x: hidden;
    position: relative;
  }
  
  html {
    scroll-behavior: smooth;
  }
  
  input {
    -webkit-appearance: none;
  }

  input:-webkit-autofill,
  input:-webkit-autofill:hover,
  input:-webkit-autofill:focus textarea:-webkit-autofill,
  textarea:-webkit-autofill:hover textarea:-webkit-autofill:focus,
  select:-webkit-autofill,
  select:-webkit-autofill:hover,
  select:-webkit-autofill:focus {
    -webkit-box-shadow: 0 0 0px 1000px #ffffff inset !important;
  }
  
  h1, h2, h3, h4, h5,h6 {
    color: ${(props) => props.theme.colors.text.headingText}
  }
  
  a {
    color: inherit;
  }
`;

export const GlobalStyles = createGlobalStyle`${globalStyles}`;
