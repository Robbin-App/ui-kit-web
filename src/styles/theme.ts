export interface IBaseRobbinTheme {
  colors: {
    primary90: string;
    primary100: string;
    primary200: string;
    secondary90: string;
    secondary100: string;
    secondary200: string;
    error: string;
    text: {
      headingText: string;
      globalText: string;
      disabledText: string;
    };
    border: {
      darkBorder: string;
      lightBorder: string;
    };
    white: string;
    greySurface: string;
  };
  shadows: {
    global: string;
    globalHover: string;
    focus: string;
    active: string;
  };
  fonts: {
    nunito: string;
    weight: {
      light: number;
      regular: number;
      semiBold: number;
      bold: number;
      extraBold: number;
    };
  };
  sizes: {
    containerMaxWidth: number;
    borderRadius: string;
  };
}

export const baseRobbinTheme: IBaseRobbinTheme = {
  colors: {
    white: '#ffffff',
    greySurface: '#FAFAFA',
    primary90: '#1A5589',
    primary100: '#15456F',
    primary200: '#133D63',
    secondary90: '#5081B9',
    secondary100: '#4677AF',
    secondary200: '#426FA4',
    error: '#A62121',
    border: {
      lightBorder: '#E7E6E6',
      darkBorder: '#D6D6D6',
    },
    text: {
      headingText: '#2E2E2E',
      globalText: '#545454',
      disabledText: '#A1A1A1',
    },
  },
  shadows: {
    global: '0px 0px 9px 2px rgba(0, 0, 0, 0.08)',
    globalHover: '0px 0px 9px 2px rgba(0, 0, 0, 0.16)',
    focus: '0px 0px 9px 2px rgba(21, 69, 111, 0.5)',
    active: '0px 0px 9px 2px rgba(21, 69, 111, 0.8)',
  },
  fonts: {
    nunito: `'Nunito', sans-serif`,
    weight: {
      light: 300,
      regular: 400,
      semiBold: 600,
      bold: 700,
      extraBold: 800,
    },
  },
  sizes: {
    containerMaxWidth: 1230,
    borderRadius: '8px',
  },
};
