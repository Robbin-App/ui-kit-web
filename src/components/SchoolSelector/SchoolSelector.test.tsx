import { ThemeProvider } from 'styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react';
import { baseRobbinTheme } from '../../styles';
import { SchoolSelector } from './index';

const renderPropComponent = (theme, props = {}) => (
  <ThemeProvider theme={theme}>
    <SchoolSelector
      name="test name"
      logoSrc="logo src"
      logoAlt="logo alt"
      {...props}
    />
  </ThemeProvider>
);

describe('SchoolSelector', () => {
  test('renders', () => {
    const render = renderer
      .create(renderPropComponent(baseRobbinTheme))
      .toJSON();
    expect(render).toMatchSnapshot();
  });
  test('calls onClick', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      renderPropComponent(baseRobbinTheme, { onClick })
    );
    fireEvent.click(getByText('test name'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
