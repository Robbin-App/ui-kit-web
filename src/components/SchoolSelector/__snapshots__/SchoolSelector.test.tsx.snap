// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolSelector renders 1`] = `
<button
  className="sc-bdnxRM llojyM"
>
  <div>
    <img
      alt="logo alt"
      src="logo src"
    />
  </div>
  <span>
    test name
  </span>
</button>
`;
